gtfs-validator/main/src/main/java/org/mobilitydata/gtfsvalidator/validator/TripHasNoDuplicateStopsValidator.java

```diff
@@ -0,0 +1,91 @@
+ * limitations under the License.
+ */
+
+package org.mobilitydata.gtfsvalidator.validator;
+
+import java.util.Arrays;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+import java.util.Objects;
+import org.mobilitydata.gtfsvalidator.annotation.GtfsValidator;
+import org.mobilitydata.gtfsvalidator.annotation.Inject;
+import org.mobilitydata.gtfsvalidator.notice.NoticeContainer;
+import org.mobilitydata.gtfsvalidator.notice.TripWithDuplicateStopNotice;
+import org.mobilitydata.gtfsvalidator.table.GtfsStop;
+import org.mobilitydata.gtfsvalidator.table.GtfsStopTableContainer;
+import org.mobilitydata.gtfsvalidator.table.GtfsStopTime;
+import org.mobilitydata.gtfsvalidator.table.GtfsStopTimeTableContainer;
+import org.mobilitydata.gtfsvalidator.table.GtfsTrip;
+import org.mobilitydata.gtfsvalidator.table.GtfsTripTableContainer;
+
+/**
+ * Validates that for each trip that consecutive stop times do not have the same stop name.
+ * It will only generate one notice per route, so if multiple trips on the same route have 
+ * the duplicate there will only be 1 notice.
+ * 
+ * [Warning]
+ *
+ * <p>Generated notice: {@link TripWithDuplicateStopNotice}.
+ */
+@GtfsValidator
+public class TripHasNoDuplicateStopsValidator extends FileValidator {
+  @Inject GtfsTripTableContainer tripTable;
+
+  @Inject GtfsStopTableContainer stopTable;
+
+  @Inject GtfsStopTimeTableContainer stopTimeTable;
+
+  @Override
+  public void validate(NoticeContainer noticeContainer) {
+    Set<Integer> seenDuplicateStops = new HashSet<>();
+    for (GtfsTrip trip : tripTable.getEntities()) {
+      final String tripId = trip.tripId();
+      List<GtfsStopTime> stopTimesForTrip = stopTimeTable.byTripId(tripId);
+
+      // Save the information about the previous stop to compare
+      String prevStopName = null;
+      String prevStopId = null;
+      long prevCsvRowNumber = -1;
+
+      // Iterate through the stops consecutively based on time
+      for (GtfsStopTime stopTime : stopTimesForTrip) {
+        GtfsStop stop = stopTable.byStopId(stopTime.stopId());
+        if (prevStopName != null) {
+          // Create the key with the route and the stop id's alphabetically
+          int combinedStopKey = -1;
+          if (prevStopId.compareTo(stop.stopId()) >= 0) {
+            combinedStopKey = Objects.hash(trip.routeId(), prevStopId, stop.stopId());
+          } else {
+            combinedStopKey = Objects.hash(trip.routeId(), stop.stopId(), prevStopId);
+          }
+          if (prevStopName.equals(stop.stopName())) {
+            // Only create the notice once for each combination of stops
+            if (!seenDuplicateStops.contains(combinedStopKey)) {
+              noticeContainer.addNotice(
+                  new TripWithDuplicateStopNotice(stop.stopName(), prevStopId, prevCsvRowNumber,
+                      stop.stopId(), stop.csvRowNumber(), trip.routeId(), trip.tripId()));
+            }
```

Comment:
How are you handling hash collision?

Reply:
In Java8 HashMap collisions are handled automatically by a balanced tree instead of a linked list for storing collided entries. Which in the worst case results in performance of O(log n) for get().

Reply:
Let's discuss this in our 1:1, I might learn something new.

Reply:
Now using autovalue to avoid using Objects.hash that could possibly still result in hash collisions.